models/models.go

```diff
@@ -0,0 +1,97 @@
+package models
+
+import (
+	"time"
+
+	"google.golang.org/protobuf/proto"
+)
+
+
+
+type EventType uint8
+
+const (
+	EventTypeCreated EventType = 1
+	EventTypeUpdated EventType = 2
+	EventTypeDeleted EventType = 3
+)
+
+// Represents a generic event with top level fields encoding occurring/generated in the system.
+type MutationEvent struct {
```

Comment:
Why did you make this public? Shouldn't consumers of this package use the `ProtoMutationEvent`
FWIW, you might consider renaming `ProtoMutationEvent` to `MutationEvent`. This makes the publically accessible model more user friendly.
```suggestion
type mutationEvent struct {
```